packages/http-server-csharp/test/generation.test.ts

```diff
@@ -625,3 +648,354 @@ it("Generates types for generic model instantiation", async () => {
     ["public partial class ToyCollectionWithNextLink"],
   );
 });
+
+it("Generates types and controllers in a service subnamespace", async () => {
+  await compileAndValidateMultiple(
+    runner,
+    `
+       using TypeSpec.Rest.Resource;
+
+       namespace MyService {
+         model Toy {
+          @key("toyId")
+          id: int64;
+      
+          petId: int64;
+          name: string;
+        }
+
+         op foo(): CollectionWithNextLink<Toy>;
+      }
+    `,
+    [
+      ["IMyServiceOperations.cs", ["interface IMyServiceOperations"]],
+      [
+        "MyServiceOperationsControllerBase.cs",
+        ["public abstract partial class MyServiceOperationsControllerBase: ControllerBase"],
+      ],
+      ["ToyCollectionWithNextLink.cs", ["public partial class ToyCollectionWithNextLink"]],
+    ],
+  );
+});
+
+it("Handles user-defined model templates", async () => {
+  await compileAndValidateMultiple(
+    runner,
+    `
+       using TypeSpec.Rest.Resource;
+
+       namespace MyService {
+         model Toy {
+          @key("toyId")
+          id: int64;
+      
+          petId: int64;
+          name: string;
+        }
+
+      @friendlyName("{name}ListResults", Item)
```

Comment:
what if the user did not define a `friendly name` for a template model?

Reply:
See this test, which addresses this precise scenario: https://github.com/microsoft/typespec/blob/main/packages/http-server-csharp/test/generation.test.ts#L608-L627

Reply:
Added a new test to cover the naming of anonymous model instantiations